sofar/sofar.py

```diff
@@ -0,0 +1,2125 @@
+                else:
+                    # create a nested list of integers and floats
+                    # separate multidimensional arrays
+                    cell = cell.split(';')
+                    cell_nd = [None] * len(cell)
+                    for idx, cc in enumerate(cell):
+                        # get rid of white spaces
+                        cc = cc.strip()
+                        cc = cc.replace(' ', ',')
+                        cc = cc.replace(' ', '')
+                        numbers = cc.split(',')
+                        cell_nd[idx] = [float(n) if '.' in n else int(n)
+                                        for n in numbers]
+
+                    cell = cell_nd
+
+                # write parsed cell to line
+                line[idc] = cell
+
+            # first line contains field names
+            if idl == 0:
+                fields = line[1:]
+                continue
+
+            # add blank comment if it does not exist
+            if len(line) == 5:
+                line.append("")
+            # convert empty defaults from None to ""
+            if line[1] is None:
+                line[1] = ""
+
+            # make sure some unusual default values are converted for json
+            if line[1] == "permute([0 0 0 1 0 0; 0 0 0 1 0 0], [3 1 2]);":
+                # Field Data.SOS in SimpleFreeFieldHRSOS and SimpleFreeFieldSOS
+                line[1] = [[[0, 0, 0, 1, 0, 0], [0, 0, 0, 1, 0, 0]]]
+            if line[1] == "{''}":
+                line[1] = ['']
+            # convert versions to strings
+            if "Version" in line[0] and not isinstance(line[1], str):
+                line[1] = str(float(line[1]))
+
+            # write second to last line
+            convention[line[0]] = {}
+            for ff, field in enumerate(fields):
+                convention[line[0]][field.lower()] = line[ff + 1]
+
+        except: # noqa
+            raise ValueError((f"Failed to parse line {idl}, entry {idc} in: "
+                              f"{file}: \n{line}\n"))
+
+    # reorder the fields to be nicer to read and understand
+    # 1. Move everything to the end that is not GLOBAL
+    keys = [key for key in convention.keys()]
+    for key in keys:
+        if "GLOBAL" not in key:
+            convention[key] = convention.pop(key)
+    # 1. Move Data entries to the end
+    for key in keys:
+        if key.startswith("Data"):
+            convention[key] = convention.pop(key)
+
+    return convention
+
+
+def _format_value_for_netcdf(value, key, dtype, dimensions, S):
+    """
+    Format value from SOFA object for saving in a NETCDF4 file.
+
+    Parameters
+    ----------
+    value : str, array like
+        The value to be formatted
+    key : str
+        The name of the current attribute. Needed for verbose errors.
+    dtype : str
+        The the data type of value
+    dimensions : str
+        The intended dimensions from ``sofa._dimensions``
```

Comment:
I'd use the getter for dimensions instead of private variables
```suggestion
        The intended dimensions from ``sofa.dimensions``
```